### PR TITLE
PE-3193: playing with enum types

### PR DESCRIPTION
**Pull request by matibat**
I don't think we're gonna merge this as it is, but we may take some pieces of it

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -6,7 +6,7 @@
     <meta name="viewport" content="width=device-width, initial-scale=1.0" />
     <title>ArNS - ar.io</title>
   </head>
-  <body>
+  <body style="margin: 0;">
```
Comment (matibat) [inline index.html:9]: I noticed we have the default body margin of 8 pixels 👀

File: src/hooks/useTransactionData/useTransactionData.tsx
Comment (matibat) [inline src/hooks/useTransactionData/useTransactionData.tsx:1]: I used here the newly implemented getSearchParam and the already-existing  isAntInteraction and isRegistryInteraction

File: src/utils/searchUtils.ts [hunk 2]
```diff
@@ -94,23 +97,11 @@ export function isInteractionCompatible({
...
-    }
-    if (
-      interactionType === ANT_INTERACTION_TYPES.UNKNOWN ||
-      interactionType === REGISTRY_INTERACTION_TYPES.UNKNOWN
-    ) {
-      throw new Error(
-        `interaction type of ${interactionType} is not compatible`,
-      );
-    }
-
```
Comment (matibat) [inline src/utils/searchUtils.ts:113]: The whole idea of having explicit types was to remove this part here and the one below lol

**Review by matibat — COMMENTED** [review]
some comments

File: src/types.ts [hunk 3]
```diff
@@ -236,15 +236,59 @@ export enum ASSET_TYPES {
...
+export type AntInteractionTypeName = (typeof antInteractionTypeNames)[number];
+export type RegistryInteractionTypeName =
+  (typeof registryInteractionTypeNames)[number];
+export type UnknownInteractionTypeName = typeof unknownInteractionTypeName;
+export type ValidInteractionTypeName =
+  | AntInteractionTypeName
+  | RegistryInteractionTypeName;
+export type InteractionTypeName =
+  | ValidInteractionTypeName
+  | UnknownInteractionTypeName;
```
[the remark this answers is not in the document]
Comment (dtfiedler) [reply src/types.ts:290]: these are great recs - cc @atticusofsparta

**Comment by dtfiedler** [comment]
Incorporated a lot of this in https://github.com/ar-io/pdns-react/pull/181

Great recs @matibat - appreciate it!

**Pull request closed without merging**